### PR TITLE
fw-4210, Songs write api

### PR DESCRIPTION
**Pull request by plasticviking**
### Description of Changes
Please briefly explain the changes here.

### Checklist
- [ ] README / documentation has been updated if needed
- [ ] PR title / commit messages contain Jira ticket number if applicable
- [ ] Tests have been updated / created if applicable
- [ ] Admin Panel has been updated if models have been added or modified
- [ ] Migrations have been updated and committed if applicable
- [ ] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [ ] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/serializers/song_serializers.py [hunk 1]
```diff
@@ -16,18 +21,50 @@ class LyricSerializer(ModelSerializer):
...
+            pass
+
+        return super().update(instance, validated_data)
+
     class Meta(SiteContentLinkedTitleSerializer.Meta):
         model = Song
+        read_only_fields = (
+            base_id_fields,
+            base_timestamp_fields,
+            "cover_image",
```
Comment (sarahfirstvoices) [inline firstvoices/backend/serializers/song_serializers.py:65]: Why is the cover_image read-only?

File: firstvoices/backend/serializers/media_serializers.py [hunk 2]
```diff
@@ -89,20 +90,44 @@ class Meta(MediaWithThumbnailsSerializer.Meta):
...
+        return VideoSerializer(context=self.context).to_representation(value)
+
+
+class WriteableRelatedImageSerializer(serializers.PrimaryKeyRelatedField):
+    def use_pk_only_optimization(self):
+        return False
+
+    def to_representation(self, value):
+        return ImageSerializer(context=self.context).to_representation(value)
+
```
Comment (sarahfirstvoices) [inline firstvoices/backend/serializers/media_serializers.py:117]: 🙌 These are nice!

Why does the image one skip the pk-only optimization?

File: firstvoices/backend/serializers/site_serializers.py [hunk 3]
```diff
@@ -82,6 +82,7 @@ class SiteDetailSerializer(SiteSummarySerializer):
     ignored_characters = SiteViewLinkField(view_name="api:ignoredcharacter-list")
     images = SiteViewLinkField(view_name="api:image-list")
     people = SiteViewLinkField(view_name="api:person-list")
+    songs = SiteViewLinkField(view_name="api:song-list")
```
Comment (sarahfirstvoices) [inline firstvoices/backend/serializers/site_serializers.py:85]: Thanks for adding this

File: firstvoices/firstvoices/settings.py [hunk 4]
```diff
@@ -262,7 +262,8 @@
 AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
 AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
 AWS_STORAGE_BUCKET_NAME = os.getenv("MEDIA_UPLOAD_S3_BUCKET")
-AWS_S3_REGION_NAME = os.getenv("MEDIA_UPLOAD_S3_REGION")
+AWS_S3_REGION_NAME = os.getenv("MEDIA_UPLOAD_S3_REGION", None)
+AWS_S3_ENDPOINT_URL = os.getenv("MEDIA_UPLOAD_ENDPOINT", None)
```
Comment (sarahfirstvoices) [inline firstvoices/firstvoices/settings.py:266]: What's this for?

File: firstvoices/backend/tests/test_apis/test_song_api.py [hunk 5]
```diff
@@ -36,7 +124,7 @@ def create_instance_with_media(
             related_videos=related_videos,
         )
 
-    def get_expected_list_response_item(self, song, site):
+    def get_expected_list_response_item_summary(self, song, site):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_apis/test_song_api.py:127]: I don't see where this is used. Are you maybe missing a test case to go with it?

**Comment by sarahfirstvoices** [comment]
Looking good! 

Still needs the new permissions for editing the Song model, but a separate PR would probably be best.

Otherwise just a couple of small questions and comments. Thanks.

File: firstvoices/backend/serializers/song_serializers.py (hunk 1 above)
Comment (plasticviking) [reply firstvoices/backend/serializers/song_serializers.py:65]: that was a remnant of a previous version. it had no effect but I've removed it for clarity.

File: firstvoices/firstvoices/settings.py (hunk 4 above)
Comment (plasticviking) [reply firstvoices/firstvoices/settings.py:266]: Local testing with Minio

File: firstvoices/backend/serializers/media_serializers.py (hunk 2 above)
Comment (plasticviking) [reply firstvoices/backend/serializers/media_serializers.py:117]: They probably all should, to be safe, and I've updated the code accordingly. If that optimization is enabled (as it is by default), the serializer receives a "stub" object that is missing properties the serializer might reference.

File: firstvoices/backend/tests/test_apis/test_song_api.py (hunk 5 above)
Comment (plasticviking) [reply firstvoices/backend/tests/test_apis/test_song_api.py:127]: yep, missed a commit there.

File: firstvoices/firstvoices/settings.py (hunk 4 above)
Comment (sarahfirstvoices) [reply firstvoices/firstvoices/settings.py:266]: Gotcha. Maybe throw a comment in there

**Review by sarahfirstvoices — APPROVED** [review]

**Pull request merged**